src/framework/configuration.rs

```diff
@@ -36,6 +37,47 @@ impl<P: Problem> Configuration<P> {
     pub fn into_builder(self) -> ConfigurationBuilder<P> {
         ConfigurationBuilder::new().do_(self.0)
     }
+
+    /// Runs the heuristic on the given problem.
+    ///
+    /// Returns the final state of the heuristic.
+    /// If the heuristic has a logger, that log can be obtained from
+    /// this state as well.
+    ///
+    /// If no random generator is provided, it will default
+    /// to a randomly seeded RNG.
+    pub fn optimize(&self, problem: &P) -> state::State {
+        let heuristic = self.heuristic();
+        let mut state = state::State::new_root();
+
+        state.insert(tracking::Log::new());
+        state.insert(Random::default());
+        state.insert(state::common::Population::<P>::new());
+
+        heuristic.initialize(problem, &mut state);
+        heuristic.execute(problem, &mut state);
+
+        state
+    }
+
+    pub fn optimize_with(
```

Comment:
Documentation is missing. Also `optimize` should mention `optimize_with`.

Reply:
Better?

Reply:
Looks great, thanks!